node/src/main/scala/coop/rchain/node/diagnostics/effects/package.scala

```diff
@@ -61,26 +76,38 @@ package object effects {
       import A._
       import Trace._
 
+      private val spans: TrieMap[TraceId, SourceTrace] = TrieMap.empty
+
       override def mark(name: String): F[Unit] = {
-        ask.flatMap {
-          case DefaultTrace(source) =>
-            trace(source) {
-              mark(name)
-            }
-          case KamonTrace(ks) => KamonTracer.mark(ks, name)
+        ask.flatMap { traceId =>
+          val k: Option[Trace] = spans.get(traceId)
+          k match {
+            case Some(st: SourceTrace) => st.mark(name)
+            case _                     => Sync[F].unit
```

Comment:
that's an error, we should `raise` it to our attention :)

Reply:
I would agree if the underlying implementation (TaskLocal) was working. for now I know it will blow up and I really don't want to throw exceptions from a measuring mechanism.

Reply:
Maybe we can log a `warn`ing in this case?